### PR TITLE
[EN] Beatmapping

### PR DESCRIPTION
**Pull request by EpicSupermind**
Lots of stuff that I'm too lazy to list here.

### CODE REVIEW

File: wiki/Beatmapping/en.md [hunk 1]
```diff
@@ -1,337 +1,229 @@
...
 
-Try and put some effort into making your sliders look nice! Lopsided/overlapped/wonky/plain-ugly sliders really ruin the look of a beatmap! Additionally, ensure that your sliders do not overlap one another, the tail end of your slider is always visible, and that the slider path is not obscured in any way - such as unpredictable U-turns or making the head and tail of the slider overlap. These slider-crafting pitfalls should be avoided!
+### Adding Spinners
 
-Adding Spinners
----------------
+To add a spinner to your map, `left-click` to begin it, then scroll the timeline to where you want the end to be, and terminate with a right-click. Try and keep your spinners to a reasonable length - overly long spinners are tiring and boring, whilst spinners that are too short are not nice. "Spinner spam" (as in, more than 5 spinners in one time) should be avoided!
```
Comment (sr229) [inline wiki/Beatmapping/en.md]: L137: Revise: Suggested ``[..] `left-click` to add the starting point for the spinner [..]``

File: wiki/Beatmapping/en.md [hunk 2]
```diff
@@ -1,337 +1,229 @@
...
-|           **Drag**           |
+[A list of appropriate custom sound filenames can be found here.](https://osu.ppy.sh/forum/t/729/start=0)
+
+### Storyboards
+
+*See also: [Storyboarding](/wiki/Storyboarding)*
+
+Click the `Design` tab to open up the [SBE (Storyboard Editor)]().
+
+Refer to [Shorcut Key Reference](wiki/Shortcut_Key_Reference)
```
Comment (sr229) [inline wiki/Beatmapping/en.md]: L229: end with period.

**Comment by sr229** [comment]
I would be also requesting someone who is proficient in Beatmapping/familiar with the editor to analyze this properly in the technical sense?

File: wiki/Beatmapping/en.md [hunk 3]
```diff
@@ -1,337 +1,236 @@
...
+===========
 
-**Beatmapping** is the process of creating beatmaps in osu! for players to play. It involves choosing a song, [timing](timing "wikilink"), placement of objects, and (optionally) [skinning](skinning "wikilink") and [storyboarding](Storyboarding "wikilink").
+**Beatmapping** is the process of creating beatmaps in *osu!* for players to play. It involves choosing a song, [timing](/wiki/Beatmap_Editor/Timing), placement of objects, and (optionally) [skinning](skinning "wikilink") and [storyboarding](Storyboarding "wikilink").
 
 To start, press the **Edit** button.
 
 Mappers usually want to get their beatmaps [ranked](Ranked_beatmap "wikilink"), which consists of a [beatmap approval](beatmap_approval "wikilink") process, where when completed successfully, a scoreboard will be assigned onto the beatmap and is considered complete unless the mapper wish to add/change something to it.
 
 The most common difficulties created in Beatmapping are:
```
Comment (TPGPL) [inline wiki/Beatmapping/en.md]: delete `created`

File: wiki/Beatmapping/en.md [hunk 4]
```diff
@@ -1,337 +1,236 @@
...
--   [Insane](Insane "wikilink")
--   [Expert](Expert "wikilink")
+-   [Easy](/wiki/Difficulties/osu!/Easy)
+-   [Normal](/wiki/Difficulties/osu!/Normal)
+-   [Hard](/wiki/Difficulties/osu!/Hard)
+-   [Insane](/wiki/Difficulties/osu!/Insane)
+-   [Expert](/wiki/Difficulties/osu!/Expert)
 
-Clicking each link will display a common set of guidelines [mappers](Mappers "wikilink") follow to be able to create an entertaining beatmap appropriate for its [difficulty](Difficulty "wikilink") level.
+Clicking each link will display a common set of guidelines [mappers](/wiki/Glossary/#beatmapper) follow to be able to create an entertaining beatmap appropriate for its [difficulty](/wiki/Difficulties) level.
```
Comment (TPGPL) [inline wiki/Beatmapping/en.md]: delete `common`;
comma before mappers (?)

File: wiki/Beatmapping/en.md [hunk 5]
```diff
@@ -1,337 +1,236 @@
...
 
-It is also not uncommon for [mappers](Mappers "wikilink") to use custom [difficulty](Difficulty "wikilink") names in place of these. (Like "Mario" for example)
+It is also not uncommon for mappers to use custom difficulty names in place of these. (Like "Mario" for example)
```
Comment (TPGPL) [inline wiki/Beatmapping/en.md]: add better example like `Extra` `Final Battle`

File: wiki/Beatmapping/en.md [hunk 6]
```diff
@@ -1,337 +1,236 @@
...
+**Timing is vital - incorrectly timed maps will not be considered for ranking.** Make sure your timing is perfect before putting any notes in. If in doubt about timing, contact a mod for help first, or submit your song into the [Beatmap Help forum](https://osu.ppy.sh/forum/10). This will save a *lot* of hassle later on.
 
 Here are some tips for getting your timing right:-
 
--   Watch the [video](http://osu.ppy.sh/forum/t/39317) [tutorials](http://osu.ppy.sh/forum/t/3815/) for basic timing methods.
+-   Watch [New mapping video tutorials](https://osu.ppy.sh/forum/t/39317) and/or [osu! Official Tutorials](https://osu.ppy.sh/forum/t/3815/) for basic timing methods.
 -   Make sure the ticks of the metronome coincide with the beat of the song for the entire length of the song. So check the timing at 5%, 25%, 50%, 75% and 100% of the song, at the very least.
 -   Take a look at the BPM gauge. If the number is very close to a whole number (until .10 difference), it is probably better (unless requested not) to round down to that whole number. If so, manually enter in the correct BPM using the Timing Setup Panel.
```
Comment (TPGPL) [inline wiki/Beatmapping/en.md]: until -> unless?

File: wiki/Beatmapping/en.md [hunk 7]
```diff
@@ -1,337 +1,236 @@
...
 -   For fine-tuning of the offset, slow the song down to 0.5x (or even 0.25x if necessary) and adjust the values until the beats exactly coincide with the metronome ticks.
--   Some songs have multiple BPMs. You can cater for this by adding a timing section (Timing --&gt; Add Timing Section) where the timing changes. The BPM and offset for the new timing section can be found exactly as before. The Timing Setup Panel gives a list of all timing sections in the song.
+-   Some songs have multiple BPMs. You can cater for this by adding a timing section (`Timing` > `Add Timing Section`) where the timing changes. The BPM and offset for the new timing section can be found exactly as before. The Timing Setup Panel gives a list of all timing sections in the song.
 
-What information do I enter in Song Setup?
-------------------------------------------
+### What information do I enter in Song Setup?
 
 You should enter this information before you create a second difficulty, to ensure that the description will be consistent across all of the maps in the set. You are currently **allowed to use Unicode characters.**
```
Comment (TPGPL) [inline wiki/Beatmapping/en.md]: enter?

File: wiki/Beatmapping/en.md [hunk 8]
```diff
@@ -1,337 +1,236 @@
...
+*See Also: [Combo](/wiki/Glossary/#combo)*
+
+Combos are groups of notes, essential to the scoring in *osu!* Different combos have different colours. It is important that you use combos in your beatmaps to ensure better perception and understanding of the song stanza. Typically, combos should be based on the sound (drumming, guitar-streaming, singing) of the music in some way so pay attention to the stanzas displayed in the timing panel if you need some guidance. As a rough guide, a good length of a combo should be around 5-15 excluding streaming (10-30).
 
 Note: End combos (Geki and Katu specifically) will increase the HP bar by a good amount compared to typical 300. Short combos are preferred since longer combos yield less health in comparison and may cause the beatmap to become unnaturally hard due to HP drain.
```
Comment (TPGPL) [inline wiki/Beatmapping/en.md]: bold  Note:

File: wiki/Beatmapping/en.md [hunk 9]
```diff
@@ -1,337 +1,236 @@
...
----------------------
+### What is Beat Spacing?
 
-<img src="Beatspacing.jpg" title="fig:A quick illustration" alt="A quick illustration" width="200" /> For intuitive play, your beatmaps should have good "beat spacing"! In general, beats that are close together on the timeline should be close together on the beatmap, and vice versa. This is to ensure the flow of the song sync with the beatmap design harmonically, making the beatmap much more enjoyable.
+![A quick illustration](Beatspacing.jpg "A quick illustration")
 
-The simplest way to ensure beat spacing is correct on your beatmaps is to utilize the editor's [Distance Snapping](Distance_Snap "wikilink") function, found in the upper-right hand side of the editor. The relative distance between notes can be adjusted using the mouse wheel. If you stick to using Distance Snap for the whole song, correct beat spacing is guaranteed!
+ For intuitive play, your beatmaps should have good beat spacing! In general, beats that are close together on the timeline should be close together on the beatmap, and vice versa. This is to ensure the flow of the song sync with the beatmap design harmonically, making the beatmap much more enjoyable.
```
Comment (TPGPL) [inline wiki/Beatmapping/en.md]: have a good

File: wiki/Beatmapping/en.md [hunk 10]
```diff
@@ -1,337 +1,236 @@
...
-----------------------------------------------------
+### Custom images/sounds
 
 -   Create your desired image/sound file.
 -   Save it with an appropriate name.
-    -   Random name: For storyboard. Does exactly nothing when not implemented onto the beatmap.
+    -   Random name for storyboard. Does exactly nothing when not implemented onto the beatmap.
         -   Example: A, Lyric1, 007, etc. (As long as you know which one is which one)
-    -   Skin name: The beatmap will use the file automatically, replacing one of the player's skin element. You will notice this.
+    -   Skin name: The beatmap will use the file automatically, replacing one of the player's skin elements. You will notice this.
```
Comment (TPGPL) [inline wiki/Beatmapping/en.md]: delete You will notice this

File: wiki/Beatmapping/en.md [hunk 11]
```diff
@@ -1,337 +1,236 @@
-What is "Beatmapping"?
-======================
+Beatmapping
+===========
 
-**Beatmapping** is the process of creating beatmaps in osu! for players to play. It involves choosing a song, [timing](timing "wikilink"), placement of objects, and (optionally) [skinning](skinning "wikilink") and [storyboarding](Storyboarding "wikilink").
+**Beatmapping** is the process of creating beatmaps in *osu!* for players to play. It involves choosing a song, [timing](/wiki/Beatmap_Editor/Timing), placement of objects, and (optionally) [skinning](/wiki/Skinning) and [storyboarding](/wiki/Storyboarding).
 
 To start, press the **Edit** button.
```
Comment (kj415j45) [inline wiki/Beatmapping/en.md]: >`` To start, press the `Edit` button. ``

File: wiki/Beatmapping/en.md [hunk 12]
```diff
@@ -1,337 +1,236 @@
...
-
--   The grid level can be changed for more precision by going to View --&gt; Grid Level. Notes will automatically snap to the grid.
-    -   You can override grid snapping and time snapping by holding Shift while performing a move/placement action.
--   Right-clicking on a note or a selected group of notes will delete them. Right-clicking elsewhere will toggle a new combo.
--   Common Windows shortcuts also apply in the osu! editor (e.g. Ctrl+A selects all notes, Ctrl+C/V is copy/paste, Ctrl+Z is undo).
+#### Artist
+
+**The band, singer, or group that performed or created the song.** *osu!* will pull the artist from the mp3 file, but please double check it to make sure it is spelled and punctuated correctly. **Anime/Game titles or companies do not belong here (they belong to `Tags`).** You can usually find the composer for anime and game songs on Wikipedia or other reliable sources such as [AnimeNewsNetwork](https://www.animenewsnetwork.com/) or [MyAnimeList](https://myanimelist.net/). Foreign names should be presented in the same name order as used on Wikipedia (generally, Given Name followed by Family Name, the western order, for consistency).
```
Comment (kj415j45) [inline wiki/Beatmapping/en.md]: ...... the `.mp3` file, ....

(?)

File: wiki/Beatmapping/en.md [hunk 13]
```diff
@@ -1,337 +1,236 @@
...
-|        **Mouse Wheel**       |
-| **Ctrl + Alt + Mouse Wheel** |
-|    **Ctrl + Mouse Wheel**    |
-|     **Alt + Mouse Wheel**    |
-|           **Drag**           |
+-   Drop it into your songs folder.
+
+[A list of appropriate custom sound filenames can be found here.](https://osu.ppy.sh/forum/t/729/start=0)
+
+### Storyboards
```
Comment (kj415j45) [inline wiki/Beatmapping/en.md]: >`### Storyboard`

**Comment by EpicSupermind** [comment]
Apple keeps doing everything i do, but better :/

**Pull request closed without merging**